lib/nitcorn/examples/src/htcpcp_server.nit

```diff
@@ -0,0 +1,90 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2015 Guilherme Mansur<guilhermerpmansur@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Hyper Text Coffee Pot Control Protocol
+#
+# A server that implements HTCPCP. At the moment there are no additions.
+#
+# The main page, `index.html`, is served dynamicly with `MyAction`.
+# The rest of the Web site fetches files from the local directory
+# `www/hello_world/`.
+import nitcorn
+
+class HTCPCPAction
+	super Action
+	var brewing = false
+
+	redef fun answer(http_request, turi) do
+		var message: String
+		var method = http_request.method
+		var headers = http_request.header
+		var response: HttpResponse
+
+		if method == "POST" or method == "BREW" then
+			if brewing then
+				message = "Pot Busy"
+				response = new HttpResponse(400)
+			else
+				message = "Brewing a new pot of coffee\n"
+				brewing = true
+				response = new HttpResponse(200)
+			end
+		else if method == "WHEN" then
+			if brewing then
+				message = "Stopped adding milk, your coffee is ready!\n"
+				brewing = false
+				response = new HttpResponse(200)
+			else
+				message = "There is no coffee brewing!"
```

Comment:
For the sake of consistency, I'd love that all messages either have a `\n` at the end of the message or not